core/src/main/scala/org/apache/spark/scheduler/ResultStage.scala

```diff
@@ -64,5 +75,16 @@ private[spark] class ResultStage(
     (0 until job.numPartitions).filter(id => !job.finished(id))
   }
 
+  def markAllPartitionsMissing(): Unit = {
+    this.discardResultsForAttemptId = this.latestInfo.attemptNumber()
+    val job = activeJob.get
+    for (id <- 0 until job.numPartitions) {
+      job.finished(id) = false
+    }
+  }
+
+  override def shouldDiscardResult(attemptId: Int): Boolean =
+    this.discardResultsForAttemptId >= attemptId
```

Comment:
Assuming both usecases for `markAllPartitionsMissing` are not required (based on discussion above), we can remove these changes.

Reply:
Unless we decide to aggressively abort the query, even if its  the first task which is failing ( & resultStage is inDeterminate),   this piece of code and the two above, are needed , IMO to avoid the race condition.